controllers/storagecluster/storageclasses.go

```diff
@@ -228,6 +228,13 @@ func newCephBlockPoolStorageClassConfiguration(initData *ocsv1.StorageCluster) S
 		disable:           managementSpec.DisableStorageClass,
 		isClusterExternal: initData.Spec.ExternalStorage.Enable,
 	}
+	if data["dataPool"] != "" {
+		sc.storageClass.Parameters = map[string]string{
```

Comment:
This creates a new map and removes what was allocated before. We just need to edit `sc.storageClass.Parameters["pool"]` and `sc.storageClass.Parameters["dataPool"]`

Reply:
right, updated accordingly.